x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/types.ts

```diff
@@ -123,11 +127,17 @@ export interface AboutStepRiskScore {
   isMappingChecked: boolean;
 }
 
+/**
+ * add / update data source types to show XOR relationship between 'index' and 'dataViewId' fields
+ * Maybe something with io-ts?
+ */
```

Comment:
We're about due to refactor these step types, so something that manages the XOR between fields on all the rule types would be a nice improvement here. Will bring it up with the team on the Rule side 👍